### PR TITLE
Use node parseArgs instead of yargs-parser and arg

### PR DESCRIPTION
**Pull request by bluwy**
## Changes

Use `node:util`'s `parseArgs` instead of `yargs-parser` and `arg` packages. There's still remnants of `yargs-parser` in `@astrojs/db` and `@astrojs/check` as the public API relies on it, but I think we can handle them later.

This PR changes quite a lot, let me know if you have any concerns and/or you feel this is worth the change.

## Testing

Tested the all CLI commands manually and working.

## Docs

I'm 99% sure this shouldn't break anything, but I wrote a changeset in case the 1% happens. It may be useful for others to know about this change as well.

### CODE REVIEW

File: packages/create-astro/src/actions/context.ts [hunk 1]
```diff
@@ -83,9 +80,24 @@ export async function getContext(argv: string[]): Promise<Context> {
...
+	no = !!no;
+	yes = !!yes;
+	install = !!install;
+	noInstall = !!noInstall;
+	git = !!git;
+	noGit = !!noGit;
+	typescript = typeof typescript == 'string' ? typescript : undefined;
+	fancy = !!fancy;
+	dryRun = !!dryRun;
+	ref = typeof ref == 'string' ? ref : undefined;
```
Comment (bluwy) [inline packages/create-astro/src/actions/context.ts:99]: One of the pain points with `parseArgs` is that if we want to allow unknown flags or positionals to be passed, it completely skips flag validation as well, so the `type` we specify for each flag is largely useless.

Anyways maybe we can consider being strict and not allow unknown options in the future, but that'll be a breaking change

**Comment by ascorbic** [comment]
Are there any warnings printed if this is used pre Node 20 when it was stabilized?

File: packages/create-astro/src/actions/context.ts (hunk 1 above)
Comment (ascorbic) [reply packages/create-astro/src/actions/context.ts:100]: Would something like `typescript ||= undefined` work here? Also I don't know what style guide we have for it, but I think an explicit `Boolean()` call is clearer than `!!`

**Comment by bluwy** [comment]
There aren't any warnings (using node 18.20.3). I'm assuming since it's stabilized on latest, the previous API (that wasn't changed) would be retroactively stable too, so I think it's safe to do so now.

File: packages/create-astro/src/actions/context.ts (hunk 1 above)
Comment (ematipico) [reply packages/create-astro/src/actions/context.ts:100]: > Anyways maybe we can consider being strict and not allow unknown options in the future, but that'll be a breaking change

IMHO, we should send this PR to `next` and break the behaviour of the CLI to report an error for unknown flags.

Comment (bluwy) [reply packages/create-astro/src/actions/context.ts:100]: > Would something like `typescript ||= undefined` work here?

The initial types for `typescript` is `string | boolean | undefined`. The line narrows it to `string | undefined`. So I'm not sure a `||=` would help here if the value is eg `true`.

> I think an explicit `Boolean()` call is clearer than `!!`

I'm team `!!` as it's shorter and [slightly faster](https://jsperf.app/woguyu) 😄 

> IMHO, we should send this PR to `next` and break the behaviour of the CLI to report an error for unknown flags.

Couldn't we do the `parseArgs` migration on main first, then do the breaking change in `next`?

**Review by ematipico — APPROVED** [review]
Looks good to me!

File: packages/create-astro/src/actions/context.ts (hunk 1 above)
Comment (bluwy) [reply packages/create-astro/src/actions/context.ts:100]: I'll take out the trash as soon as this change goes into next 😄

**Review by sarah11918 — APPROVED** [review]
For the sake of completion, approving for docs. Gotta collect the whole set!

**Review by Princesseuh — APPROVED** [review]
I'm spooked a bit by this change, but I trust you

**Pull request merged**